### ISSUE TITLE
M-Estimate Encoder 

### ISSUE THREAD

**Issue by DenisVorotyntsev** [scikit-learn-contrib/categorical-encoding#200]
I've noticed that formula for MEstimateEncoder contradicts the logic of mean target of category and also differ from formulas in [literature](http://helios.mm.di.uoa.gr/~rouvas/ssi/sigkdd/sigkdd.vol3.1/barreca.pdf): 

[m_estimate.py](https://github.com/scikit-learn-contrib/categorical-encoding/blob/master/category_encoders/m_estimate.py), line 256

`estimate = (stats['sum'] + prior * self.m) / (self._sum + self.m)`

should be replaced with 

`estimate = (stats['sum'] + prior * self.m) / (stats['count'] + self.m)`

The current formula represents something proportional to the frequency of a positive target, the second - mean target for the category. I decided to test both formulas during my benchmarking and performance of the current formula was either the same or better (I have not included it in my [repo](https://github.com/DenisVorotyntsev/CategoricalEncodingBenchmark) though). 

However, I think it should not be called M-Estimate Encoder because it doest calculated with M-Estimate formula. 

**Suggestions**: 

1.  Rename `MEstimateEncoder` to `TargetSumEncoder`
2. Add real `MEstimateEncoder`

**Comment by janmotl** [comment]
Fixed in https://github.com/scikit-learn-contrib/categorical-encoding/commit/fbadfda262e05d412bac15f553f085f5afd07a69. Thank you for reporting the bug.

> However, I think it should not be called M-Estimate Encoder because it doest calculated with M-Estimate formula.

Possibly. I used the name given to equation 7 in _A preprocessing scheme for high-cardinality categorical attributes in classification and prediction problems_. The equation 7 itself is from [On estimating probabilities in tree pruning](https://link.springer.com/chapter/10.1007/BFb0017010), where they name it _m-probability estimate_ (equation 1). Nevertheless, your proposal sounds good. What is the equation for the real m-estimate?

[](url)
